CMakeLists.txt

```diff
@@ -57,15 +56,38 @@ if (${LLVM_PACKAGE_VERSION} VERSION_LESS "5.0")
   message(FATAL_ERROR "LLVM version must be >= 5.0")
 endif()
 
+if (NOT "${CMAKE_SYSTEM_NAME}" STREQUAL "Windows")
+  if (${LLVM_PACKAGE_VERSION} VERSION_LESS "10.0")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
+  else()
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++14")
+  endif()
+endif()
+
 # Add flags for LLVM
 add_definitions(${LLVM_DEFINITIONS})
 include_directories(${LLVM_INCLUDE_DIRS})
 link_directories(${LLVM_LIBRARY_DIRS})
 
 # Get LLVM libraries for linking
-llvm_map_components_to_libnames(LLVM_LIBS
-  bitreader bitwriter core coroutines coverage instrumentation ipo irreader
-  linker lto mcparser objcarcopts option target)
+list(FIND LLVM_AVAILABLE_LIBS LLVM _LLVM_SHARED_INDEX)
+if (${_LLVM_SHARED_INDEX} GREATER -1)
+  set(LLVM_LIBS LLVM)
+else()
+  llvm_map_components_to_libnames(LLVM_LIBS
+    bitreader bitwriter core coroutines coverage instrumentation ipo irreader
+    linker lto mcparser objcarcopts option target)
+  if (${LLVM_PACKAGE_VERSION} VERSION_GREATER "9.0")
+      llvm_map_components_to_libnames(LLVM_EXTRA_LIBS frontendopenmp)
+      set(LLVM_LIBS ${LLVM_LIBS} ${LLVM_EXTRA_LIBS})
+  endif()
+endif()
+
+# https://github.com/jpalus/opencl-clang/commit/4d91f7ecdd31147e7e05ad72d0f5ec2a473f5405
```

Comment:
Can you make this a link directly to the LLVM bugzilla bug (https://bugs.llvm.org/show_bug.cgi?id=44870) please?

Reply:
Fixed.